src/Native/SymSgdNative/CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+project (SymSgdNative)
+
+set(SOURCES
+    SymSgdNative.cpp
+)
+
+if(WIN32)
+    find_library(MKL_LIBRARY MklImports HINTS ${CMAKE_SOURCE_DIR}/../../packages/mlnetmkldeps/0.0.0.5/runtimes/win-x64/native)
+else()
+    list(APPEND SOURCES ${VERSION_FILE_PATH})
+    if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+        message("Linking SymSgdNative with MKL on macOS.")
+        find_library(MKL_LIBRARY libMklImports.dylib HINTS "${CMAKE_SOURCE_DIR}/../../packages/mlnetmkldeps/0.0.0.5/runtimes/osx-x64/native")
+    else()
+        message("Linking SymSgdNative with MKL on linux.")
+        find_library(MKL_LIBRARY libMklImports.so HINTS ${CMAKE_SOURCE_DIR}/../../packages/mlnetmkldeps/0.0.0.5/runtimes/linux-x64/native)
+        SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+        SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
+        SET(CMAKE_INSTALL_RPATH "${CMAKE_SOURCE_DIR}/../../packages/mlnetmkldeps/0.0.0.5/runtimes")
```

Comment:
I'm concerned that this may work for unit tests, but it may not work on an end-user's machine.  We will have to test this as an end user to verify.

Reply:
yep. I will make sure to do this as part of bug bash.

---
In reply to: [206967492](https://github.com/dotnet/machinelearning/pull/624#discussion_r206967492) [](ancestors = 206967492)